src/dao.ts

```diff
@@ -97,14 +102,16 @@ export class DAO {
         -- all events reference an event id which contains the metadata of the event
         CREATE TABLE IF NOT EXISTS event_keys
         (
-            id                int8 GENERATED ALWAYS AS (block_number * 4294967296 + transaction_index * 65536 + event_index) STORED PRIMARY KEY,
+            id                int8 GENERATED ALWAYS AS (chain_id * 281474976710656 + block_number * 4294967296 + transaction_index * 65536 + event_index) STORED PRIMARY KEY,
```

Comment:
this is a problem:
- int8 has 63 bits to represent positive values
- we currently make the following assumptions which may not be safe if we support chains other than ethereum:
- block_number is l.t. 31 bits (2,147,483,648 blocks)
- event_index is less than 65536 (max 65k events in a single transaction)
- transaction_index is less than 4294967296 (max 4.3B transactions in a block)

we can assume transaction index instead is only 20 bits, which will add 12 bits to the block number pushing it to 8T which we should never expect to reach

Reply:
option 1